File: mysql-async/src/main/scala/com/github/mauricio/async/db/mysql/blob/encoder/ByteBufEncoder.scala
package com.github.mauricio.async.db.mysql.blob.encoder

import io.netty.buffer.ByteBuf

object ByteBufEncoder extends BlobEncoder {

  def isLong(value: Any): Boolean = value.asInstanceOf[ByteBuf].readableBytes() > BlobEncoder.LONG_THRESHOLD

  def encode(value: Any): ByteBuf = value.asInstanceOf[ByteBuf]

}

File: mysql-async/src/main/java/com/github/jasync/sql/db/mysql/binary/encoder/ByteBufEncoder.kt
package com.github.jasync.sql.db.mysql.binary.encoder

import com.github.jasync.sql.db.mysql.column.ColumnTypes
import com.github.jasync.sql.db.util.writeLength
import io.netty.buffer.ByteBuf

object ByteBufEncoder : BinaryEncoder {
    override fun encode(value: Any, buffer: ByteBuf) {
        val bytes = value as ByteBuf

        buffer.writeLength(bytes.readableBytes().toLong())
        buffer.writeBytes(bytes)
    }

    override fun encodesTo(): Int = ColumnTypes.FIELD_TYPE_BLOB
}
